core/src/avm2/globals/flash/xml/XMLNode.as

```diff
@@ -1,6 +1,82 @@
 package flash.xml
 {
    public class XMLNode {
+      internal var _isDocument: Boolean = false;
+      internal var _xml: XML;
+      internal var _parent: XMLNode = null;
+      internal var _children: Array = null;
+      internal var _index: uint = 0;
+
+      public function XMLNode(type: uint, input: String) {
+         _xml = new XML(input);
+      }
+
+      public function get attributes(): Object {
+         var attrs = new Object();
+         for each (var attribute in _xml.attributes()) {
+            attrs[attribute.localName()] = attribute.toString();
+         }
+         return attrs;
+      }
+
+      public function get childNodes(): Array {
+         // Cache the children to avoid creating new XMLNodes on
+         // every access.
+         if (_children) {
+            return _children;
+         }
+
+         var children = _xml.children();
+         var nodes = new Array();
+         for (var i = 0; i < children.length(); i++) {
+            var node = new XMLNode(0, children[i].toXMLString()); // NOPMD AvoidInstanciationInLoop
+            node._parent = this;
+            node._index = i;
+            nodes.push(node);
+         }
+         _children = nodes;
+         return nodes;
+      }
+
+      public function get firstChild(): XMLNode {
+         return childNodes[0];
+      }
+
+      public function get nextSibling(): XMLNode {
+         return this._parent.childNodes[this._index + 1];
+      }
+
+      public function get nodeName(): String {
+         if (_isDocument) {
+            return null;
+         }
+         return _xml.localName();
+      }
+
+      public function get nodeType(): uint {
+         switch (_xml.nodeKind()) {
+            case "attribute":
+            default:
```

Comment:
I think the two matches for the `switch` statement should be at the bottom of the switch statement, right? Otherwise the `default` case can be met every time and throw the Error.